vortex-datafusion/src/lib.rs

```diff
@@ -170,7 +170,7 @@ impl TableProvider for VortexMemTable {
     /// The array is flattened directly into the nearest Arrow-compatible encoding.
     async fn scan(
         &self,
-        state: &SessionState,
+        state: &dyn Session,
```

Comment:
looks like datafusion mainline changed the API here: https://github.com/apache/datafusion/pull/11516